app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -685,10 +681,21 @@ class _TextInputWidgetConsumerState extends ConsumerState<_TextInputWidget> {
   @override
   void initState() {
     super.initState();
-    WidgetsBinding.instance.addPostFrameCallback((_) {
-      widget.controller.text = ref.read(
-        chatInputProvider.select((value) => value.message),
-      );
+    ref.listenManual(
+        chatInputProvider.select((state) => state.selectedMessageState),
+        (prev, next) {
+      if (next == SelectedMessageState.edit) {
```

Comment:
there was a comment here explaining what is happening. Why did you remove that?

Reply:
I've skimmed it down to only listen ```SelectedMessageState``` events because keeping ```SelectedMessage``` is causing inconsistent frequent re-renders which resets the controller which is what we've experienced in past to avoid mixing text editing controllers with riverpod. I will bring back the comment with updated action it does.